### PR TITLE
feat(aws-stepfunctions): add support for AmazonSageMaker APIs

### LINKED ISSUE #1314: Support all Step Function task resources
At the moment, step functions only support Lambda and `Action`, it would be awesome to add support for the rest of the resources as described here: https://docs.aws.amazon.com/step-functions/latest/dg/connectors-supported-services.html:

Namely:

- [ ] AWS Batch
- [ ] Amazon DynamoDB
- [x] Amazon ECS/Fargate
- [x] Amazon SNS
- [x] Amazon SQS
- [ ] AWS Glue
- [x] Amazon SageMaker

### PR DESCRIPTION
**Pull request by mattmcclean**
Have updated the package `@aws-cdk/aws-stepfunctions-tasks` to include support for SageMaker APIs as per documentation here: https://docs.aws.amazon.com/step-functions/latest/dg/connect-sagemaker.html

Includes support for the following Amazon SageMaker API calls:
* `CreateTrainingJob`
* `CreateTransformJob`

Partially remediates #1314

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
  - __Design__: For significant features, design document added to `design` folder
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** and module name in parens, which will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+     */
+    readonly trainingJobName: string;
+
+    /**
+     * Role for thte Training Job.
+     */
+    readonly role: iam.Role;
+
+    /**
+     * Specify if the task is synchronous or asychronous.
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts]: Mention `@default`.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    /**
+     * Training Job Name.
+     */
+    readonly trainingJobName: string;
+
+    /**
+     * Role for thte Training Job.
+     */
+    readonly role: iam.Role;
```
[the remark this answers is not in the document]
Comment (rix0rrr) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts]: Also, what is the purpose of the Role?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts (hunk 1 above)
Comment (mattmcclean) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts]: added

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+import sfn = require('@aws-cdk/aws-stepfunctions');
+import { Stack } from '@aws-cdk/cdk';
+import { AlgorithmSpecification, Channel, OutputDataConfig, ResourceConfig, StoppingCondition, VpcConfig } from './sagemaker-task-base-types';
+
+export interface SagemakerTrainProps {
+
+    /**
+     * Training Job Name.
+     */
+    readonly trainingJobName: string;
```
[the remark this answers is not in the document]
Comment (mattmcclean) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts]: There is no TrainingJob resource in CFN. It is a Machine Learning job that runs for a finite time to train an ML model which eventually completes. It requires a name to be referenced and referred to.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattmcclean) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-train-task.ts]: I have updated it to be an `IRole` and will create a new Role object if not supplied. The role is a service role for Amazon SageMaker to assume to perform tasks on your behalf. It is a parameter that needs to be supplied in the SageMaker `CreateTrainingJob` API call (see docs [here](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateTrainingJob.html)). 

Also needs to be part of the IAM policy statements for the SageMaker integration (see docs [here](https://docs.aws.amazon.com/step-functions/latest/dg/sagemaker-iam.html))

**Comment by mattmcclean** [comment]
Have marked all new interfaces and classes with `@experimental`

**Pull request merged**